azavea/ism-watershed-wellness-snapshot
# Integrate Prettier

## Overview

Add Prettier to the project as well as a script to run ESLint from the command line.

While we have historically used ESLint for both code and style issues, [Prettier](https://prettier.io/) is used here to fix style issues. There is potential for conflict between the tools, and [libraries](https://github.com/prettier/eslint-config-prettier) to get around them. However, we are currently using the ESLint rules that come with CRA. The authors of CRA [decided to only have ESLint flag code issues](https://github.com/facebook/create-react-app/issues/808#issuecomment-279832535). They [recommend](https://github.com/facebook/create-react-app/issues/808#issuecomment-286870431) Prettier for code style. If we end up running into conflicts, we can add some further configuration.

Connects #8 

### Demo

![dec-14-2018 12-00-11](https://user-images.githubusercontent.com/1042475/50016678-dd22f380-ff97-11e8-8bab-78fe46aac8cf.gif)

## Testing Instructions

- On your local host, from within the `src/app` folder, run `yarn install` or `npm install` to setup the prettier commit hook.
- On the VM, run `./scripts/update`.
- Edit some of the JS files and add some badly formatted code (examples [here](https://prettier.io/playground/)).
- Try to commit the files, and verify that Prettier runs.
- After committing, verify that the edited files were cleaned up. 
- If you choose, setup your editor with Prettier. https://prettier.io/docs/en/editors.html
